ld_script_modern.txt

```diff
@@ -17,27 +17,17 @@ SECTIONS {
         gflib/*.o(ewram_data);
 
         . = 0x40000;
-}
+    }
 
     . = 0x3000000;
 
     iwram (NOLOAD) :
     ALIGN(4)
     {
-        /* .bss starts at 0x3000000 */
-        src/*.o(.bss);
-        gflib/*.o(.bss);
-        *libc.a:*.o(.bss*);
-        *libnosys.a:*.o(.bss*);
-
-        /* .bss.code starts at 0x3001AA8 */
-        src/m4a.o(.bss.code);
-
-        /* COMMON starts at 0x30022A8 */
-        src/*.o(COMMON);
-        gflib/*.o(COMMON);
-        *libc.a:*.o(COMMON);
-        *libnosys.a:*.o(COMMON);
+        *(.bss)
+        *(.bss.*)
+
+        *(COMMON)
```

Comment:
IIRC there was at least one compiler version where the explicit section definition was necessary.